### PR TITLE
initial Commit

### PR DESCRIPTION
**Pull request by webdotmilans**
Js-files of student & admin login and admin model are added

### CODE REVIEW

File: package.json
Comment (hereisadi) [inline package.json:1]: Why bcryptjs? bcrypt has already been installed.

File: src/routes/student/studentLogin.js
Comment (hereisadi) [inline src/routes/student/studentLogin.js:1]: Define the student login function in Controllers folder like this:
`const home = (req, res) => {
  res.send(
    "<p>Welcome to Vyatha api</p>"
  )
};
module.exports = {
  home,
};
`


You have to define the routes in UserRoutes.js like this:
`router.get("/", HomeController.home)`

File: src/app.js
Comment (hereisadi) [inline src/app.js:1]: You need not to do any changes in app.js file. 
create routes in UserRoutes.js file and rest will be handled automatically.
app.js has already been configured.

**Review by hereisadi — CHANGES_REQUESTED** [review]

File: src/controllers/UserController.js [hunk 1]
```diff
@@ -1,13 +1,92 @@
...
+      user: {
+        id: user.id,
+        role: 'student',
+      },
+    };
+    const authtoken = jwt.sign(data, process.env.JWT_SECRET);
+
+    res.json({ authtoken });
+  } catch (error) {
+    console.error(error.message);
```
Comment (hereisadi) [inline src/controllers/UserController.js:50]: `console.log(error)` will do the work

File: src/controllers/UserController.js [hunk 2]
```diff
@@ -1,13 +1,92 @@
...
+    res.status(500).send("Internal server error");
+  }
+};
+
+//Admin login
+
+
+const adminLogin = async (req, res) => {
```
Comment (hereisadi) [inline src/controllers/UserController.js:58]: similarly do the changes for admin login too

File: src/controllers/UserController.js [hunk 3]
```diff
@@ -1,13 +1,92 @@
...
+        id: user.id,
+        role: 'student',
+      },
+    };
+    const authtoken = jwt.sign(data, process.env.JWT_SECRET);
+
+    res.json({ authtoken });
+  } catch (error) {
+    console.error(error.message);
+    res.status(500).send("Internal server error");
```
Comment (hereisadi) [inline src/controllers/UserController.js:51]: send error message in json format.

File: src/routes/UserRoutes.js [hunk 4]
```diff
@@ -4,4 +4,8 @@ const userController = require('../controllers/UserController');
 
 router.get('/', userController.home);
 
+router.post('/vyatha/api/studentLogin',userController.studentLogin);
```
Comment (hereisadi) [inline src/routes/UserRoutes.js:7]: just give ("/studentLogin") and similary for adminLogin & add /vyatha/api is app.js

**Review by hereisadi — CHANGES_REQUESTED** [review]

**Review by hereisadi — CHANGES_REQUESTED** [review]
take pull of `dev` and solve conflicts

File: src/app.js [hunk 5]
```diff
@@ -6,6 +6,7 @@ const connectToDB = require("./db/DbConnection");
 app.use(express.json());
 
 connectToDB();
-
+app.use('/vyatha/api', userRoutes);
+module.exports = app;
```
Comment (hereisadi) [inline src/app.js:10]: redundant export.

**Review by hereisadi — CHANGES_REQUESTED** [review]

**Review by hereisadi — APPROVED** [review]

**Review by Himangshu-101 — APPROVED** [review]

**Pull request merged**